charts/nginx-gateway-fabric/values.yaml

```diff
@@ -1,4 +1,12 @@
 nginxGateway:
+  ## Set of custom annotations for GatewayClass objects, e.g:
+  ## gatewayClassCustomAnnotations:
+  ##  someKey: someValue
+  gatewayClassCustomAnnotations: {}
+  ## Set of custom annotations for NginxGateway objects, e.g:
+  ## nginxGatewayCustomAnnotations:
+  ##  someKey: someValue
+  nginxGatewayCustomAnnotations: {}
```

Comment:
I would move this field below `config`, and call it `configAnnotations`. I want to make it clear in the name that it's for the NginxGateway CRD and not the Pod (the comment for the field will help too). The naming is a little confusing now since the top level field is also called `nginxGateway`, so we would have `nginxGateway.nginxGatewayCustomAnnotations`.